### PR TITLE
[spdlog] Add spdlog support to oss-fuzz as a starting point

### PR DESCRIPTION
**Pull request by Google-Autofuzz**

### CODE REVIEW

File: projects/spdlog/project.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+homepage: "https://github.com/google/zopfli"
+primary_contact: "gmelman1@gmail.com"
+fuzzing_engines:
+  - libfuzzer
+  - afl
+sanitizers:
+  - address
+  - memory
+  -undefined
```
[the remark this answers is not in the document]
Comment (jonathanmetzman) [reply projects/spdlog/project.yaml]: I suspect this is blocking travis

File: projects/spdlog/build.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+find . -name "*.a"
+find . -name "spdlog.h"
+
+# build fuzzers
+$CXX $CXXFLAGS -std=c++11 -Iinclude \
+    $SRC/log_fuzzer.cc $LIB_FUZZING_ENGINE ./libspdlog.a \
+    -o $OUT/log_fuzzer
+
+zip $OUT/seed_corpus.zip *.*
```
Comment (jonathanmetzman) [inline projects/spdlog/build.sh]: why add every file to the seed corpus?

File: projects/spdlog/build.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+#
+################################################################################
+
+# build project
+cmake . && make -j$(nproc)
+
+find . -name "*.a"
+find . -name "spdlog.h"
+
+# build fuzzers
```
Comment (jonathanmetzman) [inline projects/spdlog/build.sh]: "fuzzer"

File: projects/spdlog/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+# limitations under the License.
+#
+################################################################################
+
+FROM gcr.io/oss-fuzz-base/base-builder
+MAINTAINER gmelman1@gmail.com
+
+RUN apt-get update && apt-get install -y make autoconf automake libtool
+RUN apt-get update && apt-get install --yes cmake
+ADD https://github.com/Kitware/CMake/releases/download/v3.14.3/cmake-3.14.3-Linux-x86_64.tar.gz $WORK
```
[the remark this answers is not in the document]
Comment (Google-Autofuzz) [reply projects/spdlog/Dockerfile]: Removed. Don't need it anymore.

File: projects/spdlog/build.sh [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+# build project
+cmake . && make -j$(nproc)
+
+find . -name "*.a"
```
[the remark this answers is not in the document]
Comment (Google-Autofuzz) [reply projects/spdlog/build.sh]: Don't need it anymore. Was looking for the library.

File: projects/spdlog/build.sh [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+find . -name "spdlog.h"
```
[the remark this answers is not in the document]
Comment (Google-Autofuzz) [reply projects/spdlog/build.sh]: Don't need it anymore.

File: projects/spdlog/project.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Google-Autofuzz) [reply projects/spdlog/project.yaml]: Got it.

File: projects/spdlog/project.yaml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+homepage: "https://github.com/google/zopfli"
+primary_contact: "gmelman1@gmail.com"
+fuzzing_engines:
```
[the remark this answers is not in the document]
Comment (Google-Autofuzz) [reply projects/spdlog/project.yaml]: Got it. Removed.

File: projects/spdlog/build.sh (hunk 2 above)
Comment (Google-Autofuzz) [reply projects/spdlog/build.sh]: Should I get rid of it if there's no corpus in spdlog yet?

File: projects/spdlog/build.sh (hunk 3 above)
Comment (Google-Autofuzz) [reply projects/spdlog/build.sh]: will use a for loop so that users can build multiple fuzzers.

File: projects/spdlog/build.sh (hunk 2 above)
Comment (jonathanmetzman) [reply projects/spdlog/build.sh]: Yes

Comment (Google-Autofuzz) [reply projects/spdlog/build.sh]: I got rid of it. At this spdlog doesn't have corpuses.

File: projects/spdlog/build.sh (hunk 3 above)
Comment (jonathanmetzman) [reply projects/spdlog/build.sh]: Then the comment should be added when the loop as added, not before hand.

**Pull request merged**